queue_services/entity-filer/src/entity_filer/filing_processors/alteration.py

```diff
@@ -41,11 +41,19 @@ def process(
     filing_meta.alteration = {}
     # Alter the corp type, if any
     with suppress(IndexError, KeyError, TypeError):
-        business_json = dpath.util.get(filing, '/alteration/business')
-        filing_meta.alteration = {**filing_meta.alteration,
-                                  **{'fromLegalType': business.legal_type,
-                                     'toLegalType': business_json.get('legalType')}}
-        business_info.set_corp_type(business, business_json)
+        if business.legal_type == Business.LegalTypes.COOP.value:
+            alteration_json = dpath.util.get(filing, '/alteration')
+            coop_association_type = alteration_json.get('cooperativeAssociationType')
+            filing_meta.alteration = {**filing_meta.alteration,
+                                      **{'fromCooperativeAssociationType': business.association_type,
+                                         'toCooperativeAssociationType': coop_association_type}}
+            business_info.set_association_type(business, coop_association_type)
+        else:
+            business_json = dpath.util.get(filing, '/alteration/business')
+            filing_meta.alteration = {**filing_meta.alteration,
+                                      **{'fromLegalType': business.legal_type,
+                                         'toLegalType': business_json.get('legalType')}}
+            business_info.set_corp_type(business, business_json)
```

Comment:
Why this is in else condition. COOP doesn't have an option to change legal_type?

Reply:
I don't think so, as per the invision designs.